server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -1320,9 +1356,9 @@ public void splitTablet(TInfo tinfo, TCredentials credentials, TKeyExtent tkeyEx
       String namespaceId;
       try {
         namespaceId = Tables.getNamespaceId(getInstance(), tableId);
-      } catch (IllegalArgumentException ex) {
-        // table does not exist, try to educate the client
-        throw new NotServingTabletException(tkeyExtent);
+      } catch (TableNotFoundException ex) {
+        // tableOperationsImpl catches ThriftSeccurityException and checks for missing table
+        throw new ThriftSecurityException(credentials.getPrincipal(), SecurityErrorCode.TABLE_DOESNT_EXIST);
```

Comment:
The client method that calls TableOperationsImpl.addSplits() has a catch for ThriftSecurityException where it checks to see if the table exists.